### PR TITLE
Possible fix to error and warning when using ROOT in pytest tests

### PR DESCRIPTION
**Pull request by guitargeek**
While testing my packages which use PyROOT, I stumbled across a little problem with ROOT and pytest which I propose to fix.

Assume you have this very minimal (basically empty) python package:
```
.
├── setup.py
└── tests
    └── test_test.py

1 directory, 2 files
```
Content of `setup.py`:
```python
from setuptools import setup, find_packages

setup(
    name="root_cleanup_test",
    packages=find_packages(),
    setup_requires=["pytest-runner"],
    test_suite="tests",
)
```
Content of `test_test.py`
```
import unittest
import ROOT

class TestTest(unittest.TestCase):
    def test_test(self):
        pass
```

Now if you run the tests with `python setup.py pytest` which gives you the warning and error I attach to the bottom of this post. I think this is maybe related to some multithreading in the pytest runner. When you run the test alone with `pytest tests/test_test.py`, you won't get the error. My setup: ROOT 6.16/00 and Python 3.7.

Let me know if I can do anything else to test this! It would be great if pytest works smoothly with ROOT.

````
========== warnings summary ==========
/usr/lib/python3.7/site-packages/ROOT.py:98: DeprecationWarning: invalid escape sequence \S
  message='class \S* already in TClassTable$' )

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========== 1 passed, 1 warnings in 0.41 seconds ==========
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/ROOT.py", line 782, in cleanup
    facade = sys.modules[ __name__ ]
KeyError: "ROOT"

### CODE REVIEW

File: bindings/pyroot/ROOT.py [hunk 1]
```diff
@@ -847,37 +847,39 @@ def cleanup():
...
-
- # destroy facade
-   facade.__dict__.clear()
-   del facade
+   if __name__ in sys.modules:
+
+      facade = sys.modules[ __name__ ]
+
+    # shutdown GUI thread, as appropriate (always save to call)
+      if hasattr( _root, 'RemoveGUIEventInputHook' ):
```
Comment (etejedor) [inline bindings/pyroot/ROOT.py]: This should probably be moved out of the `if`, since it is not related to the `facade`.

**Comment by etejedor** [comment]
Hi @guitargeek under what circumstances is `ROOT` not in `sys.modules` at cleanup time?

File: bindings/pyroot/ROOT.py (hunk 1 above)
Comment (guitargeek) [reply bindings/pyroot/ROOT.py]: You are right, I moved that above the `if`statement.

**Comment by guitargeek** [comment]
Hi @etejedor, thank's for taking care of this! I lack the technical knowledge to really isolate which circumstances lead to this error, that's why I described my use case with the python unit testing. Do you need some more details on what is exactly happening? Unfortunately I don't really have time do do some research on this this week thought...

**Comment by etejedor** [comment]
Yes, I would prefer to understand what leads to `ROOT` not being present in the system modules, is it pytest doing this?

**Comment by guitargeek** [comment]
I close this because the issue is not important for me anymore (although it still exists).

**Pull request closed without merging**